File: apps/mobile/test/contracts/binder_cover_contract_test.dart
// Contract tests for Binder tile cover pick.
//
// These assert the shared fixtures in packages/contracts, which the JavaScript
// implementation in apps/web is held to as well.
import 'package:flutter_test/flutter_test.dart';
import 'package:fabtrades/core/logic/binder_cover.dart';
import 'package:fabtrades/core/logic/binder_value_snapshot.dart';
import 'package:fabtrades/core/models/app_settings.dart';

import 'contract_fixtures.dart';

void main() {
  final contract = loadContract('binder_cover');

  for (final c in contractCases(contract, 'cases')) {
    test(c['name'] as String, () {
      final source = (c['source'] as String) == 'cardmarket'
          ? PriceSource.cardmarket
          : PriceSource.tcgplayer;
      final headline = (c['headline'] as String) == 'tcgMarketOnly'
          ? BinderValueHeadline.tcgMarketOnly
          : BinderValueHeadline.pricingValue;
      final rows = (c['entries'] as List)
          .map((e) => BinderValueRow.fromJson(Map<String, dynamic>.from(e as Map)))
          .toList();

      final cover = pickBinderCover(rows, source: source, headline: headline);
      final expected = Map<String, dynamic>.from(c['expected'] as Map);
      expect(cover.printingId, expected['printingId']);
      if (expected.containsKey('condition') && expected['condition'] != null) {
        expect(cover.condition, expected['condition']);
      }
    });
  }
}

File: apps/web/tests/contracts/binderCover.contract.test.js
// Contract tests for Binder tile cover pick.
//
// These assert the shared fixtures in packages/contracts, which the Dart
// implementation in apps/mobile is held to as well.
import contract from '../../../../packages/contracts/binder_cover.json';
import { pickBinderCover } from '../../src/utils/binderCover.js';

describe('binder cover contract', () => {
    contract.cases.forEach((testCase) => {
        it(testCase.name, () => {
            const cover = pickBinderCover(testCase.entries, {
                source: testCase.source,
                headline: testCase.headline,
            });
            expect(cover.printingId).toBe(testCase.expected.printingId);
            if (testCase.expected.condition != null) {
                expect(cover.condition).toBe(testCase.expected.condition);
            }
        });
    });
});
